src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1054,9 +1054,10 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
 
                         CorInfoType auxiliaryType = CORINFO_TYPE_UNDEF;
 
-                        if (!varTypeIsSIMD(op1->TypeGet()))
+                        if (!varTypeIsSIMD(op1))
                         {
                             auxiliaryType = CORINFO_TYPE_PTR;
+                            retNode->gtFlags |= (GTF_EXCEPT | GTF_GLOB_REF);
```

Comment:
We have to set them here manually because we don't know the auxiliary in the constructor (it is set separately here).

We could avoid this problem if we threaded the auxiliary type through the constructor and all the `gtNew` methods, but it would be lots and lots of code for this one special case.

Alternatively, we could discriminate this case by checking the `varTypeIsSIMD(Op(1))` instead of setting the auxiliary type, but I assumed this is not possible (otherwise why would this auxiliary type mechanism be used in the first place?).

Reply:
What about adding a constructor for this case or having it specially handled in `SetAuxiliaryJitType`?

Reply:
The biggest concern around having it special here, rather than in the constructor or relevant helper would be that somewhere else someone calls `gtNewSimdHWIntrinsic` for one of these intrinsics and forgets to "do the right thing".

Reply:
> What about adding a constructor for this case

The layering here is that we call the `gtNewSimdHWIntrinsicNode` method, so we'd have to add a new `gtNewSimdHWIntrinsicNode` overload as well as a new constructor for this one intrinsic, and then split the path like this:
```c
switch (intrinsicId)
{
    case OurSpecialIntrinsics:
        retNode = gtNewNewOverload(...);
        break;

    default:
       retNode = isScalar ? gtNewScalarHWIntrinsicNode(retType, op1, intrinsic)
                          : gtNewSimdHWIntrinsicNode(retType, op1, intrinsic, simdBaseJitType, simdSize);
       break;
}
```
Which is not that different from the special handling we have here already.

> The biggest concern around having it special here, rather than in the constructor or relevant helper would be that somewhere else someone calls gtNewSimdHWIntrinsic for one of these intrinsics and forgets to "do the right thing".

Indeed, but having a special constructor and `gtNewNewOverload` won't solve the problem entirely because you actually have to call them to get the handling necessary, and all existing code calls `gtNewOldOverload`, so you'd have to delete `gtNewOldOverload`, but then it is inconsistent with all other `gtNewHwi` overloads, and 99% of callsites have to pass `CORINFO_TYPE_UNDEF`, or you make it a default parameter, which defeats the purposes. I looked into this, but it did not seem like a good solution.

I think practically we're "protected enough" in the importer by the new asserts here, and after morph we have the `fgDebugCheckFlags` that will always catch it.

> having it specially handled in `SetAuxiliaryJitType`

You mean do something like "update side effect flags" inside it? I don't know, generally we don't do that (actually I am not aware of any other "setters" that would do that), the model we have is that the caller has to handle side-effect flag updates. The reason is that in the general case we can only know what flags are required once we've finished mutating the node (i. e. "setters" allow invalid intermediate states), and then `fgDebugCheckFlags` catches misconfigured nodes.